Sources/Kitura/RouterRequest.swift

```diff
@@ -102,13 +93,11 @@ public class RouterRequest {
     /// The currently matched section of the URL.
     public internal(set) var matchedPath = ""
 
-    /// The original URL as a string.
-    public var originalURL: String {
-        return serverRequest.urlString
-    }
+    /// The URL from the request if properly received
+    public internal(set) var url : URL?
 
-    /// The URL.
-    public let url: String
+    /// The URL from the request as URLComponents
+    public internal(set) var urlComponents = URLComponents()
```

Comment:
Yes, it does. If we wanted to avoid breaking changes, we could use a different name for the URL variable - `xyz: URL`, and deprecate `url: String` and `originalURL: String`, but keep them around for now. I wanted Shmuel and you to provide direction for this. Keep in mind, those variables are broken today as they return just the path of the URL(no scheme or host:port), and not the full URL as they indicate.